### ISSUE TITLE
[BUG] get_agent is working in agents.py when you pass default creds to it

### ISSUE THREAD

**Issue by vsinghal202** [GoogleCloudPlatform/dfcx-scrapi#169]
The get_agent is function in agents.py ( possibly others function might not be working) when passing default credential in current scrapi code.

## Expected Behavior
get_agent should return the agent for dfcx.

## Current Behavior
giving error:-   AttributeError: 'Credentials' object has no attribute 'universe_domain'

## Possible Solution
Error is occuring due to latest changes in upstream library https://github.com/googleapis/google-cloud-python/pull/12239. We need to update credentials based on the change.

## Steps to Reproduce

1. Create a new colab note book with code 
2.!pip install dfcx-scrapi --quiet
!pip install --upgrade gspread --quiet

import sys

if "google.colab" in sys.modules:
    from google.auth import default
    from google.colab import auth

    auth.authenticate_user()
    creds, _ = default()
else:
    # Otherwise, attempt to discover local credentials as described on https://cloud.google.com/docs/authentication/application-default-credentials
    pass

3. from dfcx_scrapi.core import agents
a = agents.Agents(creds= creds, agent_id =<agent_id>)

4.a.get_agent(agent_id)

**Comment by kmaphoenix** [comment]
After some extensive research, I've verified the root cause on this and it's not with the scrapi library. 😄 

# **_TLDR - You need to force update `google-auth>=2.27.0` for your environment_**

# Details
For Google Colab environments, the default google-auth version is `2.17.3` which is incompatible with the latest changes that were pushed ~5 days ago in [this PR](https://github.com/googleapis/google-cloud-python/pull/12239) for all Google SDKs.  
SCRAPI uses various Google SDKs under the hood.
This causes the `universe_domain` variable to fail in the workflow.

If you `pip install --upgrade google-auth`, then restart and try again, everything works fine.


I've verified this in the following environments:
- Google Colab
- Vertex Workbench notebooks
- Local VS Code Jupyter notebooks

Also with the various auth credential types:
- Local ADC
- SAK file
- SAK Dict / JSON

# SCRAPI Changes
The one change I will make to SCRAPI is to set the `google-auth` version in the `requirements.txt` file.
